draft-ietf-tls-tls13.md

```diff
@@ -2021,25 +2024,52 @@ selected_identity
   the identies in the client's list.
 {: br}
 
-If no suitable identity is provided, the server MUST NOT negotiate
-a PSK cipher suite and MAY respond with an "unknown_psk_identity"
-alert message. Sending this alert is OPTIONAL; servers MAY instead
-choose to send a "decrypt_error" alert to merely indicate an
-invalid PSK identity or instead negotiate use of a non-PSK cipher
-suite, if available.
 
-If the server selects a PSK cipher suite, it MUST send a
-"pre_shared_key" extension with the identity that it selected.
-The client MUST verify that the server's selected_identity
-is within the range supplied by the client. If the server supplies an
-"early_data" extension, the client MUST verify that the server selected the
-first offered identity. If any other value is returned, the client MUST
-generate a fatal "unknown_psk_identity" alert and close the connection.
+Each PSK offered by the client also indicates the authentication and
+key exchange modes with which the server can use it, with each
+list being in the order of the client's preference, with most
+preferred first.
+
+PskKeyExchangeModes have the following meanings:
+
+psk_ke
+: PSK-only key establishment. In this mode, the server MUST not
+supply a "key_share" value.
+
+psk_dhe_ke
+: PSK key establishment with (EC)DHE key establishment. In this mode,
+the client and servers MUST supply "key_share" values as described
+in {{key-share}}.
+{:br}
+
+PskAuthenticationModes have the following meanings:
+
+psk_auth
+: PSK-only authentication. In this mode, the server MUST NOT supply
+either a Certificate or CertificateVerify message.
+
+psk_sign_auth
+: PSK authentication plus a digital signature from the server. In this
+mode, the server MUST send Certificate ({{certificate}}) and CertificateVerify
+({{certificate-verify}}) messages.
```

Comment:
I think this wants a TON more text and clarification as to semantics and client/server policies. I'd probably even suggest we add this in as a separate change. This PR is already quite large. Right now it's primarily editorial stuff. If we also throw in PSK+Sign, then it becomes both a large bunch of editorial changes throughout the document plus thinking about some stuff like:

* What does it mean if the certificate changes?
* When should the client offer one more or another?
* How should the server decide which it prefers?
* Does this give us the overall behavior we want?

The awkward thing about PSK vs PSK+Sign compared to all our other negotiations is this varies over time. PSK+Sign's only purpose in life is to arrange for NewSessionTicket's lifetime to be uncapped. Which means that if the lifetime isn't close to expiring, both sides would prefer normal PSK just so they have to do less work (particularly servers). Only when it gets close to the client (and only the client; the server doesn't gain any direct benefit from signing again) wanting a renewal, do we want to do a PSK+Sign and switch back.